src/notation/view/internal/dynamicpopupmodel.cpp

```diff
@@ -0,0 +1,56 @@
+#include "dynamicpopupmodel.h"
+#include "src/engraving/dom/dynamic.h"
+#include "types/symnames.h"
+#include "log.h"
+
+using namespace mu::notation;
+using namespace mu::engraving;
+
+const QList<QStringList> DynamicPopupModel::DYN_NAMES = {
+    { "dynamicPP", "dynamicPiano", "dynamicMP", "dynamicMF", "dynamicForte", "dynamicFF" },
+    { "dynamicFFF", "dynamicFFFF", "dynamicFFFFF" },
+    { "dynamicFortePiano", "dynamicPF", "dynamicSforzando1", "dynamicSforzato", "dynamicSforzatoPiano" },
+    { "dynamicSforzatoFF", "dynamicSforzandoPiano", "dynamicRinforzando2", "dynamicRinforzando1", "dynamicForzando" },
+    { "dynamicPPPPPP", "dynamicFFFFFF" },
+    { "dynamicPPPPP", "dynamicPPPP", "dynamicPPP" },
+    { "dynamicCrescendoHairpin", "dynamicDiminuendoHairpin" },
+};
+
+DynamicPopupModel::DynamicPopupModel(QObject* parent)
+    : AbstractElementPopupModel(PopupModelType::TYPE_DYNAMIC, parent)
+{
+}
+
+QVariantList DynamicPopupModel::pages() const
+{
+    QVariantList variantPages;
+    for (const QStringList& page : m_pages) {
+        variantPages.append(QVariant::fromValue(page));
+    }
+    return variantPages;
```

Comment:
I think after applying my other comments, a conversion like this could also become unnecessary, namely by making `m_pages` directly a `QVariantList` (consisting of `QVariantList`s of course, but it won't be possible to tell that from type anymore; that's perhaps regrettable, but that's life, with QML…)